src/Optimization/hiopKKTLinSysMDS.cpp

```diff
@@ -527,22 +460,31 @@ namespace hiop
 			    "KKT_MDS_XYcYd linsys: MagmaNopiv size %d (%d cons) (safe_mode=%d)\n", 
 			    n, neq+nineq, safe_mode_);
 	  
-	  hiopLinSolverIndefDenseMagmaNopiv* p = new hiopLinSolverIndefDenseMagmaNopiv(n, nlp_);
-	  linSys_ = p;
-	  p->set_fake_inertia(neq + nineq);
-	}
+	         linSys_ = new hiopLinSolverIndefDenseMagmaBuKa(n, nlp_);
+	      } else {
+	        auto hovLevel = hovScalars;
+	        if(switched_linsolvers) hovLevel = hovWarning;
+
+	        nlp_->log->printf(hovLevel, 
+			      "KKT_MDS_XYcYd linsys: MagmaNopiv size %d (%d cons) (safe_mode=%d)\n", 
+			      n, neq+nineq, safe_mode_);
+ 
+          hiopLinSolverIndefDenseMagmaNopiv* p = new hiopLinSolverIndefDenseMagmaNopiv(n, nlp_);
+	        linSys_ = p;
+	        p->set_fake_inertia(neq + nineq);
```

Comment:
Again, I could use some clarification on the naming scheme for the additions

Reply:
'set_fake_inertia' comes from the legacy code. I have replaced all of them to `setFakeInertia' in the new commit.